kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/listener/CollectingTestEngineListener.kt

```diff
@@ -16,31 +19,44 @@ class CollectingTestEngineListener : AbstractTestEngineListener(), Mutex by Mute
    val names = mutableListOf<String>()
    var errors = false
 
+   /**
+    * Counts the number of times [engineStarted] and [engineFinished] have been invoked.
+    *
+    * Used by [waitForEngineFinished], so that internal Kotest tests can wait until engines
+    * are finished before performing assertions.
+    */
+   private val activeEngines = MutableStateFlow<Int?>(null)
```

Comment:
```suggestion
   private val activeEngineCount = MutableStateFlow<Int?>(null)
```

`activeEngines` suggests some collection of `Engine`s. I find that in such cases `count` helps at use sites.